### ISSUE TITLE
VS2017 MSBuild

### ISSUE THREAD

**Issue by nzbart** [ecsousa/PSVSEnv#1]
I'm using the new `vs2017` alias, but it doesn't seem to set up the correct path for MSBuild:

```powershell
C:\> vs2015
C:\> gcm msbuild | select -expand source
C:\Program Files (x86)\MSBuild\14.0\bin\MSBuild.exe
C:\> vs2017
C:\> gcm msbuild | select -expand source
C:\Windows\Microsoft.NET\Framework\v4.0.30319\MSBuild.exe
```

When I run the _Developer Command Prompt for VS 2017_, the correct version of MSBuild is found:

```cmd
C:\>powershell -noprofile -c "gcm msbuild | select -expand source"
C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\\MSBuild\15.0\bin\MSBuild.exe
```

Am I doing something wrong?

UPDATE: I was loading the VS 2015 settings in my PowerShell profile, and it appears that the VS 2017 settings don't correctly override the VS 2015 settings (as documented in the steps above). If I load a clean PowerShell instance without first setting the VS 2015 variables, the VS 2017 variables are set correctly. In summary, switching between versions in a single PowerShell instance doesn't appear to be supported, but loading VS 2017 by itself works well.

**Comment by ecsousa** [comment]
Yes, that's happening. PSVSEnv uses Visual Studio .bat files to set environment. Turns out that VS 2017 .bat file is not able to override other versions setup (it used to be up to 2015).

For instance:

```
C:\>vs2017; gcm msbuild | select -expand source
C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\\MSBuild\15.0\bin\MSBuild.exe

C:\>vs2015; gcm msbuild | select -expand source
C:\Program Files (x86)\MSBuild\14.0\bin\MSBuild.exe

C:\>vs2017; gcm msbuild | select -expand source
C:\WINDOWS\Microsoft.NET\Framework\v4.0.30319\MSBuild.exe
```

I have thought of way to save original environment setup prior to changing it, which I hope would fix the issue. But I still to implement it.

**Comment by ecsousa** [comment]
Just released version `1.1.1`, in which I've implemented a mechanism to save and restore environments variables, making VS2017 .bat file able to properly set its environment even after VS2015 have been set.

**Comment by nzbart** [comment]
Thanks!
